src/app/CommandHandler.cpp

```diff
@@ -180,63 +189,78 @@ CHIP_ERROR CommandHandler::SendCommandResponse()
 
 CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommandElement)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    CHIP_ERROR err                      = CHIP_NO_ERROR;
+    SubjectDescriptor subjectDescriptor;
     CommandPathIB::Parser commandPath;
     TLV::TLVReader commandDataReader;
-    ClusterId clusterId;
+    RequestPath requestPath;
```

Comment:
Where is ConcreteCommandPath defined? (What module/layer?) That's the issue.

Reply:
Yeah, seems ConcreteCommandPath is in app. AccessControl is at a lower layer: higher than system/platform, but lower than app and messaging. So this suggestion is fine, just can't be done unless some of these path objects are moved to lower layers. Incidentally, access control is lower than messaging because access control does not care about anything in messaging, rather messaging has to provide some subject info to access control; access control doesn't care who made it or how it was determined or attested, just that it was done correctly and can be used as received.